### ISSUE TITLE
How do I get the source text of a request that crashes when json is not correctly formatted?

### ISSUE THREAD

**Issue by budarin** [fastify/help#679]
The client sends incorrectly formatted json in the POST request and in `setErrorHandler` I get an error that `"SyntaxError: Unexpected token h in JSON at position..."`

How do I get the text of the json itself that could not be parsed?

**Comment by mcollina** [comment]
Can you provide steps to reproduce? We often need a reproducible example, e.g. some code that allows someone else to recreate your problem by just copying and pasting it. If it involves more than a couple of different file, create a new repository on GitHub and add a link to that.

**Comment by budarin** [comment]
I make a request in the browser console
 
 ```js
 const func1 = async () =>
    await (
        await fetch('/method', {
            method: 'POST',
            headers: {
                'content-type': 'application/json',
            },
            body: '{ hello: "world" }',
        })
    ).text();

void func1() ;
```

And I get an error:
`"SyntaxError: Unexpected token h in JSON at position 2\n    at JSON.parse (<anonymous>)\n    at Object.parse ...`

I would like to write the source text of the query that led to the parsing error to the log for research: `{ hello: "world" }`

**Comment by mcollina** [comment]
I don't think we are storing the unparsed value anywhere. You can customize the content-type parsing if you would like.

**Comment by budarin** [comment]
is it possible to attach the request text to the error body in case of a parsing error?

**Comment by budarin** [comment]
are there any thoughts about my recent theses?

**Comment by jsumners** [comment]
A solution was suggested to you.

**Comment by mcollina** [comment]
I don't think doing anything with the badly-formatted request body is particularly safe.
